scanner/src/main/kotlin/scanners/ScanCode.kt

```diff
@@ -89,6 +93,107 @@ class ScanCode(name: String, config: ScannerConfiguration) : LocalScanner(name,
+        @JsonProperty("scancode_notice")
+        val notice: String?,
+        @JsonProperty("scancode_version")
+        val version: Semver?,
+        @JsonProperty("scancode_options")
+        val options: Map<String, Any>?,
+        val scanStart: String?, // Not yet available in version 2.2.1.
+        val filesCount: Int?,
+
+        // This new header format is used by Scancode version 3.x.
+        val headers: List<Header>?,
+
+        val files: List<FileResult>
+    ) {
+        /**
+         * The total count of scanned files.
+         */
+        val totalFilesCount = headers.orEmpty().fold(filesCount ?: 0) { count, header ->
+            count + header.extraData.filesCount
+        }
+    }
+
+    /**
+     * The ScanCode 3.x separate header format.
+     */
+    data class Header(
+        val toolName: String,
+        val toolVersion: Semver,
+        val options: Map<String, Any>,
+        val notice: String,
+        val startTimestamp: String,
+        val endTimestamp: String,
+        val message: String?,
+        val errors: List<String>,
+        val extraData: Extra
+    )
+
+    /**
+     * The ScanCode 3.x extra header data.
+     */
+    @JsonIgnoreProperties(ignoreUnknown = true)
+    data class Extra(
+        val filesCount: Int
+    )
+
+    /**
+     * The scan result for an individual file.
+     */
+    @JsonIgnoreProperties(ignoreUnknown = true)
+    data class FileResult(
+        val path: String,
+        val scanErrors: List<String>,
+        val licenses: List<LicenseResult>?,
+        val copyrights: List<CopyrightResult>?
+    )
+
+    /**
+     * The license result for an individual file.
+     */
+    @JsonIgnoreProperties(ignoreUnknown = true)
+    data class LicenseResult(
+        val key: String,
+        val score: Float,
+        val shortName: String,
+        val category: String,
+        val spdxLicenseKey: String,
+        val startLine: Int,
+        val endLine: Int
+    ) {
+        /**
+         * The SPDX license id (or a fallback) for this license result.
+         */
+        val spdxId = spdxLicenseKey.takeUnless { it.isEmpty() }
+            // Starting with version 2.9.8, ScanCode uses "scancode" as a LicenseRef namespace, but only for SPDX
+            // output formats, see https://github.com/nexB/scancode-toolkit/pull/1307.
+            ?: if (key == "unknown") "NOASSERTION" else "LicenseRef-scancode-$key"
+    }
+
+    /**
+     * The copyright result for an individual file.
+     */
+    @JsonIgnoreProperties(ignoreUnknown = true)
+    data class CopyrightResult(
+        // While ScanCode 2.9.2 was still using "statements", version 2.9.7 is using "value".
+        val statements: List<String>?,
+        val value: String?,
+
+        val startLine: Int,
+        val endLine: Int
+    ) {
+        /**
+         * All copyright statements found in a file.
+         */
+        val allStatements = value?.let { listOf(it) } ?: statements.orEmpty()
```

Comment:
Mabye make `statements` and `value` private to prevent accidentally using them instead of `allStatements`.